app/livechat/server/lib/QueueManager.js

```diff
@@ -0,0 +1,63 @@
+import { Meteor } from 'meteor/meteor';
+import { Match, check } from 'meteor/check';
+
+import { Rooms } from '../../../models';
+import { createLivechatRoom, createLivechatInquiry } from './Helper';
+import { LivechatInquiry } from '../../lib/LivechatInquiry';
+import { callbacks } from '../../../callbacks';
+import { RoutingManager } from './RoutingManager';
+import { Livechat } from './Livechat';
+
+export const QueueManager = {
+	async requestRoom({ guest, message, roomInfo, agent }) {
+		if (!Livechat.online()) {
+			throw new Meteor.Error('no-agent-online', 'Sorry, no online agents');
+		}
+
+		check(message, Match.ObjectIncluding({
+			rid: String,
+		}));
+		check(guest, Match.ObjectIncluding({
+			_id: String,
+			username: String,
+			status: Match.Maybe(String),
+			department: Match.Maybe(String),
+		}));
+
+		const { rid } = message;
+		const name = (roomInfo && roomInfo.fname) || guest.name || guest.username;
+
+		let room;
+		let inquiry;
+
+		await this.init(rid, name, guest, message, roomInfo)
+			.then((values) => {
```

Comment:
please don't mix `await` and `then/catch` .. this could be rewritten to something like:

```javascript
try {
	const [room, inquiry] = await this.init(rid, name, guest, message, roomInfo);
} catch (e) {
	throw new Meteor.Error(error);
}